AlphaWallet/Transactions/Storage/TransactionsStorage.swift

```diff
@@ -216,7 +225,11 @@ class TransactionDataStore {
             if each.isERC20Interaction {
                 return true
             } else {
-                return realm.object(ofType: Transaction.self, forPrimaryKey: each.primaryKey) == nil
```

Comment:
I am not quite sure if this chages  could brake any logic. But its required to make transaction to be updated. @hboon any suggestions?

Reply:
Comparing `primaryKey` ensures it's the same transaction (because it has both transaction hash and chain ID).

The blockNumber comparison looks a little odd, can you explain what it is checking for? This:

```
return each.blockNumber != tx.blockNumber && each.blockNumber != 0
```

Reply:
when we sent ether from wallet A to wallet B, we add transaction to storage, added transaction is in panding state, later
when we receive callback for `FetchLatestTransactionsOperation` we receive same transaction, with updated information, updated state, block number etc.
so with current implementation we are not able to update our stored transaction with a new received tx, because we already have added tx in storage.
(as function call `filterTransactionsToNotOverrideERC20Transactions` returns an array with 0 elements)

Reply:
Ah I see. Not sure if it will break it actually. Let's just go with it and see what happens. I think it will just have a strange visual effect as described in this comment, so let's try:

>     //We pull transactions data from the normal transactions API as well as ERC20 event log. For the same transaction, we only want data from the latter. Otherwise the UI will show the cell display switching between data from the 2 source as we fetch (or re-fetch)

Reply:
ok. In the other PR we can make subscribing for tx updates, for transaction view controller and in other parts of available.